src/string_pool.cr

```diff
@@ -1,19 +1,52 @@
+# A String pool is a collection of Strings
```

Comment:
I'd like a better explanation of what is a string pool. However, I can't find the words, and the explanation you give is very simple and clear, so for now let's keep it :-)